### ISSUE TITLE
Why does naming an interface 'Props' not give correct type-checking with a functional React Native component?

### ISSUE THREAD

**Issue by nextriot** [Microsoft/tslint-microsoft-contrib#658]
### Bug Report

-   **`tslint-microsoft-contrib` version**: ^5.2.1
-   **TSLint version**: ^5.11.0
-   **TypeScript version**: ^3.1.6
-   **Running TSLint via**: VS Code

#### TypeScript code being linted

```tsx
import * as React from 'react'
import { View } from 'react-native'

interface Props {
  name: string
  color: string
  price: string
}

const Card = ({ name, color, price }: Props) => (
  <View style={{ backgroundColor: color }} />
)

export default Card

```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended",
    "tslint-config-standard",
    "tslint-react",
    "tslint-config-prettier"
  ],
  "jsRules": {},
  "rulesDirectory": ["node_modules/tslint-microsoft-contrib"],
  "rules": {
    "ordered-imports": false,
    "object-literal-sort-keys": false,
    "member-ordering": false,
    "jsx-no-lambda": false,
    "jsx-boolean-value": false,
    "interface-name": false,
    "semicolon": false,
    "react-unused-props-and-state": [
      true,
      { "props-interface-regex": "Props$", "state-interface-regex": "State$" }
    ]
  }
}

```

#### Actual behavior

Typescript does not pick up on the fact that the color property has been used in the component, and reports: `[tslint] Unused React property defined in interface: name [react-unused-props-and-state]`, at the interface level.

However, if the functional component is changed to a class component, or simply, if the interface is named something other than Props e.g. CardProps - it picks up on the property's usage as expected.

#### Expected behavior

I would expect the compiler to pick up on the fact that the property is being used within the component, as it does in the other scenarios mentioned above.

**Comment by JoshuaKGoldberg** [comment]
Shouldn't TypeScript's `--noUnusedParameters` flag _(or TSLint's `no-unused-variable` if you're still using it) pick up on this instead?

**Comment by IllusionMH** [comment]
Options are:
- Disable this rule until it is fixed (I'd propose this one).

- Disable this rule on file by file basis with `// tslint:disable react-unused-props-and-state`  comment
(only if have a ton of class components (that do not use destructuring on `this.props`/`this.state`) and can benefit from this rule in general)

- Use different naming conventions for class and function in a way so props type name won't match regexp for class components props type name
(worst option because this will introduce unneeded differences in naming, another thing to remember)
   

Generally you should stick with rules that allow you to avoid errors/mistakes, enforce code style or, be more productive.

`react-unused-props-and-state` supposed to catch potential mistakes, but in its current state produces false positives for all functional components and may negatively affect productivity.

**Comment by JoshuaKGoldberg** [comment]
Closing as a duplicate of #339. Thanks for the discussion folks!
